app/services/backdate_parser.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+class BackdateParser
+  def initialize(date:)
+    @raw_date = date.to_h
+    @issue_items = []
+  end
+
+  def issues
+    Requirements::CheckerIssues.new(issue_items)
+  end
+
+  def parse
+    check_date_is_valid
+    return if issues.any?
+
+    Time.current.change(
+      day: raw_date[:day].to_i,
+      month: raw_date[:month].to_i,
+      year: raw_date[:year].to_i,
+      hour: 0,
+      minute: 0,
+    )
```

Comment:
It might be cleaner to have this as a Date object at this point in the code rather than a Time object. 

This might also make this class simpler as we already create a Date object in line 32.

It would mean having to then convert the date to a time in the interactor, but I think that would actually be better as that feels a more appropriate place to set a magic time on it (I believe the requested magic time is 1 minute past midnight)

Reply:
Do we have any rationale behind setting the time to 00:01?  I think it would be helpful to include this in the commit message to explain it.

Reply:
Not a great bit of rationale, could be worth asking Tobi for more info on that tomorrow - I think it's based on an expectation that users get confused by 00:00 (but that never confuses me so I don't relate) - there's this comment about it: https://trello.com/c/aBHXR82t/932-user-can-backdate-or-set-document-as-new#comment-5cf8dbb949a3fa4dfbe0291d

Reply:
Discussed with Tobi and since the time isn't going to be shown in Content Publisher or on the frontend, we're going to stick with midnight.

Reply:
I think we should use 10:10 ;-).

https://petapixel.com/2013/06/27/why-photographs-of-watches-and-clocks-show-the-time-1010/